### ISSUE TITLE
HashTreeRoot panics in validator first block

### ISSUE THREAD

**Issue by prestonvanloon** [prysmaticlabs/prysm#2959]
Blocking #2312 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x84f449]

goroutine 12 [running]:
github.com/prysmaticlabs/go-ssz.makeFieldsHasher.func1(0xc0a3c0, 0xc000288a50, 0x199, 0x0, 0x0, 0x0, 0x0, 0x0, 0x203000, 0x203000)
        external/com_github_prysmaticlabs_go_ssz/hash_tree_root.go:310 +0x689
github.com/prysmaticlabs/go-ssz.truncateAndHash(0xc0a3c0, 0xc000288a50, 0x199, 0x0, 0x0, 0x0, 0x0, 0x30, 0xc000288a50)
        external/com_github_prysmaticlabs_go_ssz/signing_root.go:46 +0x1f1
github.com/prysmaticlabs/go-ssz.SigningRoot(0xc35680, 0xc000288a50, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000337800)
        external/com_github_prysmaticlabs_go_ssz/signing_root.go:27 +0x1c4
github.com/prysmaticlabs/prysm/validator/client.(*validator).AttestToBlockHead(0xc000350090, 0xd7ec20, 0xc0003368d0, 0x0, 0xc0000411a0, 0x60)
        validator/client/validator_attest.go:98 +0xb90
github.com/prysmaticlabs/prysm/validator/client.run.func1(0xd87f80, 0xc000350090, 0xd7ebe0, 0xc00012b260, 0x0, 0xc000000001, 0xc0000411a0, 0x60)
        validator/client/runner.go:83 +0x3d2
created by github.com/prysmaticlabs/prysm/validator/client.run
        validator/client/runner.go:80 +0x6c8

```

**Comment by prestonvanloon** [comment]
It looks like the validator is looking to take the signing root of a struct that does not have a signature field.

https://github.com/prysmaticlabs/prysm/blob/0acf164ebd275e91e325db6aae0dbbe8ef45a8b0/validator/client/validator_attest.go#L93-L98

**Comment by prestonvanloon** [comment]
Changing this from signing root to tree hash root resolves the problem, although we should still fix the panic and consider if this is the correct API / usage.

**Comment by terencechain** [comment]
Looks like this is fixed in #2960
